pkg/client/client.go

```diff
@@ -212,6 +221,15 @@ func Initialize(
 				}
 				defer eventDeduplicator.NotifyKeyGenCompleted(event.KeepAddress)
 
+				keep, err := ethereumChain.GetKeepWithID(event.KeepAddress)
+				if err != nil {
+					logger.Errorf(
+						"failed to resolve keep with address [%v] for created event: [%v]",
```

Comment:
In general it shouldn't matter. But, this and the bad log output you flagged above are both due to an issue with how go-ethereum handles formatting that was fixed in [v1.9.25](https://github.com/ethereum/go-ethereum/commit/bd848aad7c4e1f7d1eaecd9ea7ee23785090768a). We're trailing pretty hard, so I'm wondering if it's worth checking to see if a 1.9.25 bump would break everything, but I don't think the refactor should be blocked on it.

We're also about to swap all this out with the *ID types, which should handle stringification properly. But we're also merging this to master, so I'm a little torn about what to do heh.

Reply:
Will we have the code`*ID` types ready soon? Because if so, we can maybe merge one PR right after another. 🤷

Reply:
This week I think. I have a PR coming with the tBTC Application refactor, and then I think the IDs will be next.